### PR TITLE
Develop - Merge to TOT Master

### PR DESCRIPTION
**Pull request by kiritigowda**
* rocAL updates

### CODE REVIEW

**Comment by LakshmiKumar23** [comment]
@kiritigowda Need to resolve #1037 before merging this

**Comment by kiritigowda** [comment]
@LakshmiKumar23 there are a few rocAL/ADAT codacy issues, can you see if you can fix/ignore them? -- https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=11161368&bid=33744141

**Comment by kiritigowda** [comment]
@LakshmiKumar23 - this PR contains rocAL changes which fail on OpenCL. Need to fix them before merging. 
```
 61%] Building CXX object amd_openvx_extensions/amd_rpp/CMakeFiles/vx_rpp.dir/source/SequenceRearrange.cpp.o
/var/jenkins_home/workspace/_Short-GPUOpen_MIVisionX_PR-1039@2/MVQMWSbnD/OpenCL/mivisionx/amd_openvx_extensions/amd_rpp/source/ResizeMirrorNormalizeTensor.cpp:90:1: error: function definition is not allowed here
{
^
/var/jenkins_home/workspace/_Short-GPUOpen_MIVisionX_PR-1039@2/MVQMWSbnD/OpenCL/mivisionx/amd_openvx_extensions/amd_rpp/source/ResizeMirrorNormalizeTensor.cpp:130:1: error: function definition is not allowed here
{
^
/var/jenkins_home/workspace/_Short-GPUOpen_MIVisionX_PR-1039@2/MVQMWSbnD/OpenCL/mivisionx/amd_openvx_extensions/amd_rpp/source/ResizeMirrorNormalizeTensor.cpp:156:1: error: function definition is not allowed here
{
^
/var/jenkins_home/workspace/_Short-GPUOpen_MIVisionX_PR-1039@2/MVQMWSbnD/OpenCL/mivisionx/amd_openvx_extensions/amd_rpp/source/ResizeMirrorNormalizeTensor.cpp:247:1: error: function definition is not allowed here
{
^
/var/jenkins_home/workspace/_Short-GPUOpen_MIVisionX_PR-1039@2/MVQMWSbnD/OpenCL/mivisionx/amd_openvx_extensions/amd_rpp/source/ResizeMirrorNormalizeTensor.cpp:281:1: error: function definition is not allowed here
{
^
/var/jenkins_home/workspace/_Short-GPUOpen_MIVisionX_PR-1039@2/MVQMWSbnD/OpenCL/mivisionx/amd_openvx_extensions/amd_rpp/source/ResizeMirrorNormalizeTensor.cpp:293:1: error: function definition is not allowed here
{
^
/var/jenkins_home/workspace/_Short-GPUOpen_MIVisionX_PR-1039@2/MVQMWSbnD/OpenCL/mivisionx/amd_openvx_extensions/amd_rpp/source/ResizeMirrorNormalizeTensor.cpp:337:2: error: expected '}'
}
 ^
/var/jenkins_home/workspace/_Short-GPUOpen_MIVisionX_PR-1039@2/MVQMWSbnD/OpenCL/mivisionx/amd_openvx_extensions/amd_rpp/source/ResizeMirrorNormalizeTensor.cpp:56:1: note: to match this '{'
{
^
7 errors generated when compiling for gfx90a.
make[2]: *** [amd_openvx_extensions/amd_rpp/CMakeFiles/vx_rpp.dir/build.make:986: amd_openvx_extensions/amd_rpp/CMakeFiles/vx_rpp.dir/source/ResizeMirrorNormalizeTensor.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 61%] Building CXX object amd_openvx_extensions/amd_rpp/CMakeFiles/vx_rpp.dir/source/SnowbatchPD.cpp.o
make[1]: *** [CMakeFiles/Makefile2:426: amd_openvx_extensions/amd_rpp/CMakeFiles/vx_rpp.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 61%] Linking CXX shared library ../../lib/libvx_opencv.so
[ 61%] Built target vx_opencv
make: *** [Makefile:156: all] Error 2
script returned exit code 2
```

@rrawther @swetha097 @SundarRajan28 
See logs for more information->
http://math-ci.rocm.amd.com/blue/organizations/jenkins/compute-rocm-dkms-no-npi-hipclang%2FShort-GPUOpen%2FMIVisionX/detail/PR-1039/13/pipeline

image

**Comment by SundarRajan28** [comment]
Created a PR #1049 for fixing the OCL build issue

**Comment by LakshmiKumar23** [comment]
@kiritigowda can we merge this? QA needs it for testing

**Comment by kiritigowda** [comment]
@LakshmiKumar23 can you look at these codacy errors from rocAL - https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pull/1039/checks?check_run_id=12288019969

File: docker/tensorflow/mivisionx-with-tensorflow.dockerfile [hunk 1]
```diff
@@ -55,4 +55,8 @@ RUN git clone https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX
 RUN cd ~/models/research/ && \
 protoc object_detection/protos/*.proto --python_out=. && \
 cp object_detection/packages/tf2/setup.py . && \
-pip3 install --use-feature=2020-resolver .
+python -m pip install .
+
+#upgrade pip
+RUN pip3 install --upgrade pip
+
```
Comment (LakshmiKumar23) [inline docker/tensorflow/mivisionx-with-tensorflow.dockerfile:62]: @kiritigowda can you change this to the version you used in the setup script. That should fix this issue

File: rocAL/rocAL_pybind/examples/tf_petsTrainingExample/requirements.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# installing required packages
+pip install contextlib2
```
Comment (LakshmiKumar23) [inline rocAL/rocAL_pybind/examples/tf_petsTrainingExample/requirements.sh:2]: @kiritigowda Can you add this line in the PR itself as line 1 in the file. Should fix the issue
```
#!/bin/bash
```

**Comment by LakshmiKumar23** [comment]
> @LakshmiKumar23 can you look at these codacy errors from rocAL - https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pull/1039/checks?check_run_id=12288019969

@kiritigowda please see comments below

**Pull request merged**